### ISSUE TITLE
Allow multiple sources for "extends" config parameter

### ISSUE THREAD

**Issue by DethAriel** [palantir/tslint#1487]
### Enhancement Request

- __TSLint version__: `3.14.0`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: CLI

`"extends"` config parameter should allow an *array* of "base" configurations to be passed.

#### Reasoning

There are custom rule sets created by the community. Some of them are fairly large (like [tslint-microsoft-contrib](https://github.com/Microsoft/tslint-microsoft-contrib)) and contain a pretty big number of custom rules. It might get difficult to keep up with all of them, and even track which rule is taken from which set. Plus, such a mechanism will allow rule-providers to write their own `"*:latest"` and 
`"*:recommended"` rule sets.

#### Expected usage

```json
{
  "extends": [
    "tslint:latest",
    "tslint-microsoft-contrib:latest (not that it's supported, but it could be)",
    "some-npm-module-with-main",
    "./my-custom-overrides.json"
  ]
}
```

If there are multiple values for the same rule in different sources, the latest will be taken (the `"./my-custom-overrides.json"` in the above example). The `"rules"` section in the `tslint.json` file itself will still override whatever is set in the `"extends"`.

**Comment by adidahiya** [comment]
I think this already works -- did you try it @DethAriel?

**Comment by adidahiya** [comment]
In particular, [this code](https://github.com/palantir/tslint/blob/81890b61d8c4a246e2c3d141f93f449d8e4fa221/src/configuration.ts#L139) manages merging of sources in the `extends` config

**Comment by DethAriel** [comment]
I tried it one more time - and I can't get it work with [codelyzer](https://github.com/mgechev/codelyzer) rules in a separate file. I'll take a look tomorrow and provide an update

**Comment by DethAriel** [comment]
@adidahiya , that doesn't look to work afterall. 

##### `test.ts`
```ts
export { thing: "my-stuff" };
```
##### `tslint.json`
```json
{
  "extends": ["tslint:latest", "./mytslint.json"],
  "rules": {}
}
```
##### `mytslint.json`
```json
{
  "quotemark": [ true, "single" ]
}
```
##### `tsconfig.json`
```json
{
  "compilerOptions": {
    "target": "es5",
    "module": "commonjs"
  },
  "files": ["./test.ts"]
}
```

#### Command
`tslint --project ./tsconfig.json`

#### Output:
* No lint errors

#### Expected Output:
* `test.ts[1, 17]: " should be '`

#### Comments
Everything works as expected if the `tslint.json` file looks like this:
```json
{
  "extends": ["tslint:latest", "./mytslint.json"],
  "rules": {  
    "quotemark": [ true, "single" ]
  }
}
```

**Comment by adidahiya** [comment]
`mystlint.json` is malformed. you need to include the `"rules"` key:

```json
{
  "rules": {
    "quotemark": [ true, "single" ]
  }
}
```

**Comment by DethAriel** [comment]
@adidahiya here's a [unit test](https://github.com/DethAriel/tslint/commit/74e458f3e1d87c23842fc00eb6e139a52dd4b3d7) that proves it's not working:

**Comment by JKillian** [comment]
@DethAriel, the issue here, I believe, is that the "extends" configurations are applied in reverse order to what you are expecting. The last config you list in the `extends` array is actually the "base" config, and then each preceding configuration is applied on top of it.

In your case, `"extends": ["./mytslint.json", "tslint:latest"],` should work.

**Comment by DethAriel** [comment]
@JKillian yes, thanks, that's what I thought too when browsing the sources. Yet this looks a bit counter-intuitive. But it might be that I don't see some trivial concept behind this.

Since this appears to be working after all, maybe it would make sense to add arrayish `"extends"` to the documentation along with the reasoning on why does the "inheritance" work this way instead of the other. Thoughts, @adidahiya ?

**Comment by adidahiya** [comment]
@DethAriel yep, definitely, better docs around this would definitely be accepted as contributions

**Comment by JKillian** [comment]
@DethAriel Yes, I may have made the wrong choice on the order. Your way does seem a little more intuitive, although I'm worried about changing things now...

**Comment by DethAriel** [comment]
@JKillian I see your point, that would be a breaking change. Since there's little left to do on the initial issue, I'll open a different one with more context
